### PR TITLE
Ai refactor

### PR DESCRIPTION
**Pull request by Will-D-AER**

### CODE REVIEW

File: src/seu_injection/metrics/accuracy.py [hunk 1]
```diff
@@ -0,0 +1,543 @@
...
+    # Check if x_tensor is actually a DataLoader
+    if hasattr(x_tensor, "__iter__") and hasattr(x_tensor, "dataset"):
+        # It's a DataLoader, use the loader function
+        if y_true is not None:
+            # TODO ERROR HANDLING: Inconsistent exception types across framework
+            # ISSUE: Mixed use of ValueError, RuntimeError, TypeError without clear patterns
+            # CURRENT: ValueError for user input errors, but RuntimeError for device issues
+            # IMPROVEMENT: Define custom exception hierarchy with clear usage guidelines
+            # EXAMPLES: SEUInputError, SEUDeviceError, SEUComputationError
+            # PRIORITY: MEDIUM - affects user experience and debugging efficiency
```
[the remark this answers is not in the document]
Comment (Will-D-AER) [reply src/seu_injection/metrics/accuracy.py]: This can be added as a TODO

File: src/seu_injection/core/injector.py [hunk 2]
```diff
@@ -0,0 +1,610 @@
...
+
+                    # Inject fault, evaluate, restore
+                    tensor.data[idx] = torch.tensor(
+                        seu_val, device=self.device, dtype=tensor.dtype
+                    )
+                    criterion_score = self.get_criterion_score()
+                    tensor.data[idx] = original_tensor[idx]  # Restore original value
+
+                    # Record results
+                    results["tensor_location"].append(idx)
```
[the remark this answers is not in the document]
Comment (wd7512) [reply src/seu_injection/core/injector.py]: This could be useful as a TODO @copilot can you add this as a todo comment?
